torch/csrc/jit/codegen/cuda/shape_inference.cpp

```diff
@@ -492,18 +483,43 @@ class NaiveTypePropagator {
           ? *scalar_type
           : c10::promoteTypes(*op0->scalarType(), *op1->scalarType());
 
-      return TensorType::create(
-          promoted_scalar_type, *op0->device(), c10::nullopt, c10::nullopt);
+      if (op0->isComplete()) {
+        return TensorType::create(
+            promoted_scalar_type,
+            *op0->device(),
+            op0->sizes(),
+            op0->strides(),
+            c10::nullopt);
+      } else if (op1->isComplete()) {
```

Comment:
:red_square: <- red flagging.
This is bug prone.
Missing shape information on `op0` doesn't guarantee output to be casted to shape of `op1`.

Reply:
So, we should infer the output shape only when shape information is present for both arguments?

Reply:
Yes I think so. We try our best to infer the right shape, otherwise, we should just leave it as nullopt and assert later when it's not available